### PR TITLE
Add Lightning Watchtower TEOS + add zmq fragment

### PR DESCRIPTION
**Pull request by Kukks**
https://github.com/talaia-labs/python-teos

@RiccardoMasutti  @sr-gi  can you test?

### CODE REVIEW

**Comment by sr-gi** [comment]
Yep, will do.

File: docker-compose-generator/docker-fragments/bgold-lnd.yml [hunk 1]
```diff
@@ -31,21 +31,11 @@ services:
 
   btcpayserver:
     environment:
-      BTCPAY_BTCLIGHTNING: "type=lnd-rest;server=http://lnd_bitcoingold:8080/;macaroonfilepath=/etc/lnd_bitcoingold/admin.macaroon;allowinsecure=true"
+      BTCPAY_BTGLIGHTNING: "type=lnd-rest;server=http://lnd_bitcoingold:8080/;macaroonfilepath=/etc/lnd_bitcoingold/admin.macaroon;allowinsecure=true"
```
Comment (sr-gi) [inline docker-compose-generator/docker-fragments/bgold-lnd.yml:34]: I'm not familiar with the codebase, but this looks like a typo (`BTG`). Ignore if not.

Comment (Kukks) [reply docker-compose-generator/docker-fragments/bgold-lnd.yml:34]: this was actually a bug with some altcoin misconfiguring stuff if activated

Comment (sr-gi) [reply docker-compose-generator/docker-fragments/bgold-lnd.yml:34]: Ohh wait, bitcoingold... Nevermind.

File: docker-compose-generator/docker-fragments/opt-add-zmq.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+version: "3"
+
+services:
+  bitcoind:
+    environment:
+      BITCOIN_EXTRA_ARGS: |
+        zmqpubrawblock=tcp://0.0.0.0:28332
```
Comment (sr-gi) [inline docker-compose-generator/docker-fragments/opt-add-zmq.yml:7]: For `teos` we also need `zmqpubhashblock=tcp://0.0.0.0:28332`

File: docker-compose-generator/docker-fragments/opt-add-teos.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      BTC_RPC_CONNECT: "bitcoind"
+      BTC_RPC_PORT: 43782
+      BTC_RPC_USER: "teos"
+      BTC_RPC_PASSWORD: "teos"
+      BTC_FEED_CONNECT: "bitcoind"
+      BTC_FEED_PORT: 28332
+    volumes:
+      - "bitcoin_teos_datadir:/root/.teos"
+    ports:
+      - "9814"
```
Comment (sr-gi) [inline docker-compose-generator/docker-fragments/opt-add-teos.yml:20]: should be `export 9814:9814` unless there's any good reason to not map it to that port.

Comment (Kukks) [reply docker-compose-generator/docker-fragments/opt-add-teos.yml:20]: afaik saying "9814" maps to "9814:9814"

File: docker-compose-generator/docker-fragments/opt-add-zmq.yml (hunk 2 above)
Comment (Kukks) [reply docker-compose-generator/docker-fragments/opt-add-zmq.yml:7]: We can't do this, we have `zmqpubrawtx` mapped to a different port that other services already use

File: docker-compose-generator/docker-fragments/opt-add-teos.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    environment:
+      API_BIND: "0.0.0.0"
+      API_PORT: 9814
+      BTC_NETWORK: ${NBITCOIN_NETWORK:-regtest}
+      BTC_RPC_CONNECT: "bitcoind"
+      BTC_RPC_PORT: 43782
+      BTC_RPC_USER: "teos"
+      BTC_RPC_PASSWORD: "teos"
+      BTC_FEED_CONNECT: "bitcoind"
+      BTC_FEED_PORT: 28332
```
Comment (Kukks) [inline docker-compose-generator/docker-fragments/opt-add-teos.yml:16]: any way to split this to two properties for tx and block zme endpoints? @sr-gi

File: docker-compose-generator/docker-fragments/opt-add-teos.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        # rpcpassword=teos
+        # We need to use rpcauth because we also need cookieauth. rpcpassword disabled cookie file auth.
+        # Be careful if you copy the line below from the docker-compose.yml! A dollar sign is escaped.
+        rpcauth=teos:504ff14fe34faa90e5bbe124e94be6a4$$bdac33da901ff27b4a27ca160cd706b2ba41d9e363556815cab5b7ef1e4dcf23
+volumes:
+  bitcoin_teos_datadir:
+
+required:
+  - "opt-add-zmq"
+  - "bitcoin"
```
Comment (Kukks) [inline docker-compose-generator/docker-fragments/opt-add-teos.yml:34]: @sr-gi  does teos need tx-index to work?

File: docker-compose-generator/docker-fragments/opt-add-zmq.yml (hunk 2 above)
Comment (sr-gi) [reply docker-compose-generator/docker-fragments/opt-add-zmq.yml:7]: `zmqpubhashblock` you mean?

If so, teos only needs `zmqpubhashblock`, so mapping it to that one would do.

File: docker-compose-generator/docker-fragments/opt-add-teos.yml (hunk 5 above)
Comment (sr-gi) [reply docker-compose-generator/docker-fragments/opt-add-teos.yml:34]: Yeah, it actually does. I Forgot to mention that.

File: docker-compose-generator/docker-fragments/opt-add-teos.yml (hunk 3 above)
Comment (sr-gi) [reply docker-compose-generator/docker-fragments/opt-add-teos.yml:20]: It was not, if not specifying `9814:9814` I was getting it mapped to `32877`

**Comment by RiccardoMasutti** [comment]
I have not found any particular problems but I think we should find other testers

**Comment by sr-gi** [comment]
Testing this one last time on amd64 and arch32.

**Comment by Kukks** [comment]
> I have not found any particular problems but I think we should find other testers

Have you run it on an actual server and connected to the watch tower?

**Comment by sr-gi** [comment]
There was an issue with the python3 version used by the entrypoint script at least on amd64. I've updated that and re-uploaded the docker images to dockerhub.

Tested for amd64 and arch32, looks good.

PS: Currently looks like dockerhub is catching the old latest as latest so the error may persist for a while. I've tried it manually getting latest and it work fine.

**Comment by NicolasDorier** [comment]
@Kukks you don't have to update the shitcoins... but well

**Comment by sr-gi** [comment]
Thanks to both 🚀

**Pull request merged**